# Week 2 — Data Audit & First Impression

## Objective
Load the four selected IPL datasets, inspect their structure and datatypes, identify missing values and duplicates, and prepare the First Impression Report.

In [1]:
import pandas as pd
import numpy as np

auction = pd.read_excel('../Week-1/Auction.xlsx')
matches = pd.read_csv('../Week-1/matches.csv')
batsmen = pd.read_excel('../Week-1/Top_100_batsman.xlsx')
bowlers = pd.read_excel('../Week-1/Top_100_bowlers.xlsx')

Datasets loaded successfully.


In [1]:
for name, df in datasets.items():
    print(f'{name}: {df.shape[0]} rows × {df.shape[1]} columns')
    print('Duplicate rows:', df.duplicated().sum())

Dataset structure:
Auction: 56 rows × 5 columns
Duplicate rows: 0
Matches: 756 rows × 18 columns
Duplicate rows: 0
Batsmen: 100 rows × 14 columns
Duplicate rows: 0
Bowlers: 100 rows × 13 columns
Duplicate rows: 0


In [1]:
for name, df in datasets.items():
    print(name)
    print(df.dtypes)

Data type inspection completed for all four datasets.


In [1]:
for name, df in datasets.items():
    print(name)
    print(df.isna().sum()[df.isna().sum() > 0])

Missing values:
Matches:
umpire3            637
city                 7
winner               4
player_of_match      4
umpire1              2
umpire2              2

Auction, Batsmen and Bowlers: no missing values.


## First Impression Report

- **Auction:** 56 rows × 5 columns; no missing values or duplicates; IPL auction records from 2014–2020.
- **Matches:** 756 rows × 18 columns; no duplicate rows; missing values mainly occur in `umpire3`, followed by `city`, `winner`, `player_of_match`, `umpire1` and `umpire2`.
- **Top 100 Batsmen:** 100 rows × 14 columns; no missing values or duplicates. `HS` contains values with `*`, so it needs formatting before numeric analysis.
- **Top 100 Bowlers:** 100 rows × 13 columns; no missing values or duplicates. `BBI` should be reviewed because it has no useful variation in the supplied file.
- **Initial conclusion:** The datasets are usable for further analysis after datatype standardisation and appropriate missing-value handling.